### PR TITLE
Cleanup `HomeScreen` and `MediaPage`

### PR DESCRIPTION
**Pull request by imashnake0**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

We need to cleanup `Home` and `MediaPage` since they haven't been touched in ages and because we need to create a `BannerLayout` for `ProfileScreen`.

**Summary of changes:**

1. Cleanup `HomeScreen` and `MediaPage`.

**Tested changes:**

[comment]: # (If changes were tested, replace this comment with a brief description of how you did so. Else replace it with "N/A".)

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

File: app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt [hunk 1]
```diff
@@ -76,12 +77,15 @@ fun Home(
     val upcomingList by viewModel.upcomingMediaNextSeason.collectAsState()
     val allTimePopularList by viewModel.allTimePopular.collectAsState()
 
-    // TODO: [Code Smells: If Statements](https://dzone.com/articles/code-smells-if-statements).
+    val homeRowListToHeader = listOf(
+        trendingList to stringResource(R.string.trending_now),
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt:81]: Can't we create categories from the ViewModel or even deeper in the arch?

File: app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt [hunk 2]
```diff
@@ -279,13 +274,15 @@ fun MediaDetails(
...
-                    TextView(it).apply {
+                factory = { context ->
+                    TextView(context).apply {
                         movementMethod = LinkMovementMethod.getInstance()
                         setTextColor(textColor)
                         textSize = 14f
                         // This is needed since `FontFamily` can't be used with `AndroidView`.
-                        typeface = ResourcesCompat.getFont(it, com.imashnake.animite.core.R.font.manrope_medium)
+                        typeface = ResourcesCompat.getFont(
+                            context, com.imashnake.animite.core.R.font.manrope_medium
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt:284]: Full qualified import? 😢

Comment (imashnake0) [reply app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt:284]: true

File: app/src/main/java/com/imashnake/animite/features/home/HomeViewModel.kt [hunk 3]
```diff
@@ -94,5 +95,9 @@ class HomeViewModel @Inject constructor(
 
     companion object {
         const val NOW = "now"
+        const val TRENDING_NOW = "Trending Now"
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/home/HomeViewModel.kt:98]: I guess string-ify is coming later?

File: app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt [hunk 4]
```diff
@@ -76,12 +78,15 @@ fun Home(
     val upcomingList by viewModel.upcomingMediaNextSeason.collectAsState()
     val allTimePopularList by viewModel.allTimePopular.collectAsState()
 
-    // TODO: [Code Smells: If Statements](https://dzone.com/articles/code-smells-if-statements).
+    val rows = listOf(
+        trendingList,
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt:82]: I think I preferred "string" to trendingList 😔

**Review by boswelja — APPROVED** [review]

**Pull request merged**